apps/charterafrica/src/lib/ecosystem/ecosystem/processData.js

```diff
@@ -1,24 +1,76 @@
+import github from "@/charterafrica/lib/ecosystem/github";
 import {
   createCollection,
   bulkMarkDeleted,
   getCollectionIdsPerAirtableId,
 } from "@/charterafrica/lib/ecosystem/payload";
+import api from "@/charterafrica/lib/payload";
 import {
   ORGANIZATION_COLLECTION,
   CONTRIBUTORS_COLLECTION,
   TOOL_COLLECTION,
 } from "@/charterafrica/payload/utils/collections";
 
-export async function processContributors(airtableData, config) {
+async function toolFromCacheOrGit(airtableData, update) {
+  if (update) {
+    return github.fetchTool(airtableData);
+  }
+  const { airtableId } = airtableData;
+  const { docs } = await api.getCollection(TOOL_COLLECTION, {
+    where: {
+      airtableId: { equals: airtableId },
+    },
+  });
+  if (docs.length) {
+    return docs[0];
+  }
+  return github.fetchTool(airtableData);
+}
+
+async function organisationFromCacheOrGit(airtableData, update) {
+  if (update) {
+    return github.fetchOrganisation(airtableData);
+  }
+  const { airtableId } = airtableData;
+  const { docs } = await api.getCollection(ORGANIZATION_COLLECTION, {
+    where: {
+      airtableId: { equals: airtableId },
+    },
+  });
+  if (docs.length) {
+    return docs[0];
+  }
+  return github.fetchOrganisation(airtableData);
```

Comment:
I'm still not sure about this @koechkevin:

1. If we're updating, fetch fresh data from GitHub (Called from `updateContent`). ✅ 
2. If we're not updating, I'm assuming we're creating/preparing data i.e. called from `updatedList`. Yet, if `airtableId` is missing from collection, we're again fetching from GitHub. ❓❓❓ . Should we query GitHub **ONLY** for entities that we are sure are in the CMS collection already?

Reply:
@kilemensi 
The reason  why I had it this way is 
When `updateList` is called and the  data already exists in the collection, then we only use it to link during preparation. No need to make an API call at this point.

Reply:
Do you suggest that we don't make any github api call from `updateList` only to be done by `updateContent`?

Reply:
Yes... `updateList` will be called more often than `updateContent` I believe.

Reply:
Yes,

But if `updateList` is called and new Items are added then it means for the period between calling `updateList` and `updateContent` the newly added items will have github data missing @kilemensi

Reply:
That's alright for now... we have different API limits on Airtable and GitHub so we can't assume just because we can pull from Airtable, we can also pull from GitHub @koechkevin .

As long as missing data doesn't break the system, we'll experiment & figure out what's the best way to call these 2 methods.